mlir/test/Dialect/Luminous/kernel-outlining.mlir

```diff
@@ -50,16 +50,47 @@ module {
     return
   }
 }
-// CHECK-LABEL: luminous.module @device_module
-// CHECK-LABEL: luminous.func @async_fn_1
-// CHECK-LABEL: luminous.func @async_fn_2
-// CHECK-LABEL: luminous.func @async_fn_3
+
+// CHECK: luminous.module @[[CAPSULE:.*]]
+// CHECK: luminous.func @[[ASYNC_FN_0:.*]]
+// CHECK: luminous.func @[[ASYNC_FN_1:.*]]
+// CHECK: luminous.func @[[ASYNC_FN_2:.*]]
 // CHECK-LABEL: func.func @test2
     // CHECK: (%[[ARG0:.*]], %[[ARG1:.*]], %[[ARG2:.*]], %[[ARG3:.*]])
     // CHECK: %[[C1024:.*]] = arith.constant 1024 : index
     // CHECK: luminous.launch shape (%[[C1024]]) step (%[[C1024]])
     // CHECK: ^[[BLOCK:.*]](%[[ARG4:.*]]):
-        // CHECK: %[[DISP0:.*]] = luminous.dispatch  @device_module::@async_fn_1 (%[[ARG0:.*]], %[[ARG1:.*]], %[[ARG2:.*]])
-        // CHECK: %[[DISP1:.*]] = luminous.dispatch  [%[[DISP0]]] @device_module::@async_fn_2 (%[[ARG0:.*]], %[[ARG2:.*]], %[[ARG1:.*]])
-        // CHECK: %[[DISP2:.*]] = luminous.dispatch  [%[[DISP1]]] @device_module::@async_fn_3 (%[[ARG1:.*]], %[[ARG2:.*]], %[[ARG3:.*]])
-        // CHECK: async.await %[[DISP2]] : !async.token
\ No newline at end of file
+        // CHECK: %[[DISP0:.*]] = luminous.dispatch  @[[CAPSULE:.*]]::@[[ASYNC_FN_0:.*]] (%[[ARG0:.*]], %[[ARG1:.*]], %[[ARG2:.*]])
+        // CHECK: %[[DISP1:.*]] = luminous.dispatch  [%[[DISP0]]] @[[CAPSULE:.*]]::@[[ASYNC_FN_1:.*]] (%[[ARG0:.*]], %[[ARG2:.*]], %[[ARG1:.*]])
+        // CHECK: %[[DISP2:.*]] = luminous.dispatch  [%[[DISP1]]] @[[CAPSULE:.*]]::@[[ASYNC_FN_2:.*]] (%[[ARG1:.*]], %[[ARG2:.*]], %[[ARG3:.*]])
+        // CHECK: async.await %[[DISP2]] : !async.token
+
+
+// -----
+
+module {
+  func.func @test3(%arg0: memref<1024x1024xf32>, %arg1: memref<1024xf32>, %arg2: memref<1024xf32>) {
+    %c1024 = arith.constant 1024 : index
+    %c0 = arith.constant 0 : index
+    luminous.launch shape (%c1024) step (%c1024){
+    ^bb0(%arg4: index):
+      scf.for %arg5 = %c0 to %c1024 step %c1024 {
+        linalg.matvec {"linalg-max-memory-footprint" = 10000 : i64} ins(%arg0, %arg1 : memref<1024x1024xf32>, memref<1024xf32>) outs(%arg2 : memref<1024xf32>)
+      }
+      luminous.yield
+    }
+    return
+  }
+}
+
+// CHECK: luminous.module @[[CAPSULE:.*]]
+// CHECK: luminous.func @[[ASYNC_FN_0:.*]]
```

Comment:
should we be more cautious and actually check that the `linalg.matvec` made it in here?

Reply:
updated. added one more test with nested forloops